### ISSUE TITLE
Duplicated mac addresses in a zone

### ISSUE THREAD

**Issue by weizhouapache** [apache/cloudstack#7633]
Note: This issue is difficult to reproduce,but we do notice it in one environment, which caused some issues.

When create a VM on isolated network, the mac address is generated by the following method
```
    @Override
    public String getNextAvailableMacAddress(final long networkConfigId, Integer zoneMacIdentifier) {
        final SequenceFetcher fetch = SequenceFetcher.getInstance();
        long seq = fetch.getNextSequence(Long.class, _tgMacAddress, networkConfigId);
        if(zoneMacIdentifier != null && zoneMacIdentifier.intValue() != 0 ){
            seq = seq | _prefix << 40 | (long)zoneMacIdentifier << 32 | networkConfigId << 16 & 0x00000000ffff0000l;
        }
        else {
            seq = seq | _prefix << 40 | _rand.nextInt(Short.MAX_VALUE) << 16 & 0x00000000ffff0000l;
        }
        return NetUtils.long2Mac(seq);
    }
```

It could happen that the mac addresses are duplicated in the same network or among different networks.
Ideally the mac address should be unique in a zone.

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~

~~~

##### CLOUDSTACK VERSION

~~~

~~~

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by DaanHoogland** [comment]
I found four places in which MAC addressses are created:

1. https://github.com/apache/cloudstack/blob/main/engine/orchestration/src/main/java/org/apache/cloudstack/engine/datacenter/entity/api/db/dao/EngineDataCenterDaoImpl.java#L148
2. https://github.com/apache/cloudstack/blob/main/engine/schema/src/main/java/com/cloud/dc/dao/DataCenterDaoImpl.java#L239
3. https://github.com/apache/cloudstack/blob/main/engine/schema/src/main/java/com/cloud/network/dao/NetworkDaoImpl.java#L433
4. https://github.com/apache/cloudstack/blob/main/server/src/main/java/com/cloud/network/NetworkModelImpl.java#L594

I think we need to check in each of these cases

**Comment by DaanHoogland** [comment]
I found four places in which MAC addressses are created:

1. https://github.com/apache/cloudstack/blob/main/engine/orchestration/src/main/java/org/apache/cloudstack/engine/datacenter/entity/api/db/dao/EngineDataCenterDaoImpl.java#L148
2. https://github.com/apache/cloudstack/blob/main/engine/schema/src/main/java/com/cloud/dc/dao/DataCenterDaoImpl.java#L239
3. https://github.com/apache/cloudstack/blob/main/engine/schema/src/main/java/com/cloud/network/dao/NetworkDaoImpl.java#L433
4. https://github.com/apache/cloudstack/blob/main/server/src/main/java/com/cloud/network/NetworkModelImpl.java#L594

I think we need to check in each of these cases.

**Comment by DaanHoogland** [comment]
> it seems the method `getNextAvailableMacAddressPair` is never used

no, and it is implemented twice, I'll remove those in the PR.

**Comment by weizhouapache** [comment]
fixed by #7634
